compiler/src/iree/compiler/Dialect/Stream/Transforms/FoldUniformOperands.cpp

```diff
@@ -35,6 +36,54 @@ namespace {
 // Per-dispatchable export optimization
 //===----------------------------------------------------------------------===//
 
+// Returns the uniformly duplicated values across all indexing map entries of
+// duplicated values and a bit vector indicating the 'dead' values.
+//
+// Example:
+// Map of duplicated values per entry: [[0, 1, 0, 1], [0, 1, 0, 2]]
+// Returns:
+// - A map of uniformly duplicated values: [0, 1, 0, 2]
+// - A bit vector with dead values: b0010
+static std::pair<SmallVector<unsigned>, llvm::BitVector>
+getUniformDupeIndexingAndDeadValues(
+    const SmallVector<SmallVector<unsigned>> &dupeIndexMaps) {
+  if (dupeIndexMaps.size() == 0) {
+    return {};
+  }
+  unsigned numValues = dupeIndexMaps[0].size();
+  llvm::BitVector sameValues(numValues);
+  llvm::BitVector deadOperandsMap(numValues);
+  SmallVector<unsigned> uniformDupeIndexMap =
+      llvm::to_vector(llvm::seq(0u, numValues)); // old -> new
+  for (unsigned idx = 0; idx < numValues; ++idx) {
+    if (deadOperandsMap.test(idx))
+      continue;
```

Comment:
style nit: add braces. I understand that this is a refactoring, but it'd be good to apply the simple fix in the refactoring.

```suggestion
    if (deadOperandsMap.test(idx)) {
      continue;
    }
```